app/services/requirement_form_json_service.rb

```diff
@@ -340,6 +342,91 @@ def get_multi_contact_datagrid_form_json(requirement_block_key = requirement&.re
     }
   end
 
+  def get_nested_info_component(field_key, parent_key, label, required, field_type = nil)
+    key = snake_to_camel(field_key.to_s)
+    key = "#{parent_key}|#{key}" if parent_key.present?
+
+    if field_type.present?
+      raise Error if DEFAULT_FORMIO_TYPE_TO_OPTIONS[field_type.to_sym].blank?
+      {
+        key: key,
+        type: field_type,
+        requirementInputType: field_type,
+        tableView: true,
+        input: true,
+        label: label,
+        widget: {
+          type: "input",
+        },
+      }.merge!(DEFAULT_FORMIO_TYPE_TO_OPTIONS[field_type.to_sym])
+    else
+      {
+        applyMaskOn: "change",
+        tableView: true,
+        input: true,
+        key: key,
+        label: label,
+        type: "textfield",
+        validate: {
+          required: required,
+        },
+        **DEFAULT_FORMIO_TYPE_TO_OPTIONS[:text],
+      }
+    end
+  end
+
+  def get_pid_info_components(requirement_block_key = requirement&.requirement_block&.key)
+    return {} unless requirement.input_type_pid_info?
+    # key = "#{requirement.key(requirement_block_key)}|pid_info?"
+    component = {
+      legend: requirement.label,
+      key: requirement_block_key,
+      type: "fieldset",
+      custom_class: "contact-field-set",
+      label: requirement.label,
+      hideLabel: true,
+      input: false,
+      tableView: false,
+      components: [
+        get_columns_form_json(
+          "pid_entry_columns",
+          [
+            get_nested_info_component(:pid, requirement_block_key, "PID", false),
+            get_nested_info_component(:folio_number, requirement_block_key, "Folio Number", false),
+            get_nested_info_component(:address, requirement_block_key, "Address", false, :address),
+          ],
+        ),
+      ],
+    }
+    multi_data_grid_from_json(requirement_block_key, component, true, "Add #{requirement.label}")
+  end
+
+  #this is a generic mulitgrid for a single component
+  def multi_data_grid_from_json(
+    override_key,
+    component,
+    initEmpty = true,
+    addMoreText = I18n.t("formio.requirement.multi_grid.default_add")
+  )
+    {
+      label: "Additional #{requirement.label}(s)",
+      id: requirement.id,
+      reorder: false,
+      addAnother: addMoreText,
+      addAnotherPosition: "bottom",
+      layoutFixed: false,
+      enableRowGroups: false,
+      initEmpty: initEmpty,
+      hideLabel: false,
+      tableView: false,
+      custom_class: "contact-data-grid",
```

Comment:
make sure if this class is needed. Seems to be specific for contact types

Reply:
I've left it for now, the styling is the same.